plugins/catalog-graphql/src/service/client.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity } from '@backstage/catalog-model';
+import fetch from 'node-fetch';
+
+export class CatalogClient {
+  constructor(private baseUrl: string) {}
+  async list(): Promise<Entity[]> {
+    const res = await fetch(`${this.baseUrl}/catalog/entities`);
```

Comment:
hm yeah it depends on what we do about the discovery stuff but I feel the `/catalog` part should be part of the constructor path; this class should assume that it is given the FULL url to a catalog endpoint and root off of that